test/e2e/e2e_test.go

```diff
@@ -1252,6 +1253,254 @@ var _ = Describe("e2e non-vxlan external gateway and update validation", func()
+
+		// Create the pod that will be used as the source for the connectivity test
+		createGenericPod(f, srcPingPodName, workerNodeInfo.name, f.Namespace.Name, command)
+		// wait for pod setup to return a valid address
+		err = wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
+			pingSrc, err = getPodAddress(srcPingPodName, f.Namespace.Name)
+			if err != nil {
+				framework.Logf("retrying ... error trying to get pod  %s address: %v", srcPingPodName, err)
+				return false, nil
+			}
+			validIP := net.ParseIP(pingSrc)
+			if validIP == nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		// fail the test if no address is ever retrieved
+		if err != nil {
+			framework.Failf("Error trying to get the pod IP address")
+		}
+		// add a host route on the gateways for return traffic to the pod
+		_, err = runCommand("docker", "exec", gwContainer1, "ip", "route", "add", pingSrc, "via", nodeIP)
+		if err != nil {
+			framework.Failf("failed to add the pod host route on the test container %s: %v", gwContainer1, err)
+		}
+		_, err = runCommand("docker", "exec", gwContainer2, "ip", "route", "add", pingSrc, "via", nodeIP)
+		if err != nil {
+			framework.Failf("failed to add the pod host route on the test container %s: %v", gwContainer2, err)
+		}
+		// install tcpdump to check for egress traffic to the gateway's loopback address
+		// since /proc/net/dev counters only record the ingress interface of the traffic
+		// a packet capture is needed to ensure traffic reaches the gateway loopback
+		_, err = runCommand("docker", "exec", gwContainer1, "yum", "-y", "install", "tcpdump")
+		if err != nil {
+			framework.Failf("failed to install tcpdump on container %s: %v", gwContainer1, err)
+		}
+		_, err = runCommand("docker", "exec", gwContainer2, "yum", "-y", "install", "tcpdump")
+		if err != nil {
+			framework.Failf("failed to install tcpdump on container %s: %v", gwContainer2, err)
+		}
+
+		// Verify the gateways and remote loopback addresses are reachable from the pod.
+		// Iterate checking connectivity to the loopbacks on the gateways until tcpdump see
+		// the traffic or 20 attempts fail. Odds of a false negative here is ~ (1/2)^20
+		ginkgo.By(fmt.Sprintf("Verifying ecmp connectivity to the external gateways by iterating through the prefix %s", exGWRemoteIpPrefix))
+
+		// validate the first gateway receives icmp packets destined for the loopback
+		for lastOctet := 1; lastOctet <= ecmpRetry; lastOctet++ {
+			gwLo := fmt.Sprintf("%s%d", exGWRemoteIpPrefix, lastOctet)
+			// spawn a goroutine to asynchronously ping the first gateway's loopbacks
+			go func() {
+				_, err = framework.RunKubectl("exec", srcPingPodName, frameworkNsFlag, testContainerFlag, "--", "ping", "-w", "20", gwLo)
+				if err != nil {
+					framework.Logf("error generating a ping from the test pod %s: %v", srcPingPodName, err)
+				}
+			}()
+		}
+		// wait until an ICMP packet is matched on the 1st gw or fail the test if one is never received
```

Comment:
you could simply this a bit, by launching the tcpdumps first as separate goroutines, then launch the ping in the main thread here. You can have the tcpdump goroutines write a bool back to a channel when if they get receive a ping. Then after your ping in this main thread you can simply read off of the channel to see if you have 2 trues.

Reply:
Ty! Was on the fence on this on adding more complexity, but I think you're spot on. Added a channel to check for any error returned from the `timeout tcpdump`. I still ran the ICMPs in a go routine to fire them off async since we just care that one hits on each gateway instead of blocking for one to hit both gateways.